control-plane/grpc/src/operations/frontend_node/client.rs

```diff
@@ -0,0 +1,122 @@
+use crate::{
+    context::{Client, Context, TracedChannel},
+    frontend::{
+        frontend_node_grpc_client::FrontendNodeGrpcClient,
+        registration::registration_client::RegistrationClient, GetFrontendNodeRequest,
+    },
+    operations::{
+        frontend_node::traits::{
+            to_list_frontend_node_request, FrontendNodeInfo, FrontendNodeOperations,
+            FrontendNodeRegisterInfo,
+        },
+        Pagination,
+    },
+};
+use stor_port::{
+    transport_api::{v0::FrontendNodes, ReplyError, ResourceKind, TimeoutOptions},
+    types::v0::{
+        openapi::client::Uri,
+        transport::{Filter, FrontendNode, MessageIdVs},
+    },
+};
+
+/// RPC Frontend Node Client.
+#[derive(Clone)]
+pub struct FrontendNodeClient {
+    ops: Client<FrontendNodeGrpcClient<TracedChannel>>,
+    registration: Client<RegistrationClient<TracedChannel>>,
+}
+
+impl FrontendNodeClient {
+    /// Creates a new base tonic endpoint with the timeout options and the address.
+    pub async fn new<O: Into<Option<TimeoutOptions>> + Clone>(addr: Uri, opts: O) -> Self {
+        let ops_client = Client::new(addr.clone(), opts.clone(), FrontendNodeGrpcClient::new).await;
+        let registration_client = Client::new(addr, opts, RegistrationClient::new).await;
+        Self {
+            ops: ops_client,
+            registration: registration_client,
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl FrontendNodeOperations for FrontendNodeClient {
+    async fn get(&self, req: Filter, ctx: Option<Context>) -> Result<FrontendNode, ReplyError> {
+        let request = self.ops.request(
+            GetFrontendNodeRequest::try_from(req.clone())?,
+            ctx,
+            MessageIdVs::GetFrontendNodes,
+        );
+        let response = self
+            .ops
+            .client()
+            .get_frontend_node(request)
+            .await?
+            .into_inner()
+            .frontend_node;
```

Comment:
Could the proto reply just be FrontendNode? or is it because we want to support returning other things?